flink-core/src/main/java/org/apache/flink/api/common/eventtime/CombinedWatermarkStatus.java

```diff
@@ -66,25 +66,25 @@ public void add(PartialWatermark element) {
     public boolean updateCombinedWatermark() {
         long minimumOverAllOutputs = Long.MAX_VALUE;
 
-        boolean hasOutputs = false;
+        // if we don't have any outputs minimumOverAllOutputs is not valid, it's still
+        // at its initial Long.MAX_VALUE state and we must not emit that
+        if (partialWatermarks.isEmpty()) {
+            this.idle = true;
```

Comment:
`this.idle = combinedWatermark > Long.MIN_VALUE`

Reply:
good idea!